ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -107,10 +108,18 @@ void handleStatement(
       final Command command,
       final CommandId commandId
   ) {
+    final CommandStatusFuture statusFuture;
+    synchronized (statusFutures) {
+      // Get a future for the in-flight request that this command may be for
+      // Once we've got a reference to it, we can clean it up from the map.
+      statusFuture = statusFutures.get(commandId);
+      statusFutures.remove(commandId);
```

Comment:
Replace these two with `statusFuture = statusFutures.remove(commandId);`

Reply:
Ack. Hard to shake my c++ conditioning sometimes :)